src/components/Forms/Textarea.jsx

```diff
@@ -1,13 +1,14 @@
 import clsx from "clsx";
 import PropTypes from "prop-types";
 import React, { forwardRef } from "react";
+import TextareaAutosize from "react-textarea-autosize";
 import { BaseInput } from "./BaseInput";
 
-export const Textarea = forwardRef(({ className, value, rows = 2, ...rest }, ref) => {
+export const Textarea = forwardRef(({ className, value, shouldAutoSize = false, rows = 2, ...rest }, ref) => {
     return (
         <BaseInput
             ref={ref}
-            as="textarea"
+            as={shouldAutoSize ? TextareaAutosize : "textarea"}
```

Comment:
Please add a link to documentation for this. Here's my problem: 

1. `TextareaAutosize` has different props than what we use for `Textarea`. So if you pass in new props, validation will fail (a giant console error) because it's not defined in `Textarea.propTypes`
2. The `rows` prop you pass, does not work with `TextareaAutosize`. It always has the same row count.

https://www.npmjs.com/package/react-textarea-autosize